spv/verifypayment.go

```diff
@@ -79,45 +126,46 @@ func (v *verifier) verifyTxAnchor(ctx context.Context, payment *Envelope) (bool,
 	// If the txid of the Merkle Proof doesn't match the txid provided in the spv.Envelope,
 	// fail and error
 	if proofTxID != payment.TxID {
-		return false, errors.Wrapf(ErrTxIDMismatch, "envelope tx id %s does not match proof %s", payment.TxID, proofTxID)
+		return errors.Wrapf(ErrTxIDMismatch, "envelope tx id %s does not match proof %s", payment.TxID, proofTxID)
 	}
 
 	valid, _, err := v.VerifyMerkleProofJSON(ctx, payment.Proof)
 	if err != nil {
-		return false, err
+		return err
 	}
-
-	return valid, nil
+	if !valid{
+		return ErrInvalidProof
```

Comment:
Could this error be wrapped, to have the id of the tx with the invalid proof?

Reply:
Sorted